monkey/infection_monkey/network/relay/utils.py

```diff
@@ -51,18 +54,14 @@ def _check_if_island_server(server: str) -> bool:
     logger.debug(f"Trying to connect to server: {server}")
 
     try:
-        requests.get(  # noqa: DUO123
-            f"https://{server}/api?action=is-up",
-            verify=False,
-            timeout=MEDIUM_REQUEST_TIMEOUT,
-        )
+        HTTPIslandAPIClient(server)
```

Comment:
We need to figure out what we actually want to do here. This code creates an instance of a concrete implementation of `IIslandAPIClient` and doesn't do anything else with it.

We may need to modify `find_server` to do the following:
- Accept a factory that constructs concrete instances of `IIslandAPIClient`
- Return an instance of IIslandAPIClient on success

Reply:
@ilija-lazoroski is working on it as we speak.